beets/importer.py

```diff
@@ -946,18 +952,19 @@ def choose_match(self, session):
     def reload(self):
         self.item.load()
 
-    def set_fields(self):
+    def set_fields(self, lib):
         """Sets the fields given at CLI or configuration to the specified
-        values.
+        values, for the singleton item.
         """
-        for field, view in config['import']['set_fields'].items():
-            value = view.get()
-            log.debug(u'Set field {1}={2} for {0}',
-                      displayable_path(self.paths),
-                      field,
-                      value)
-            self.item[field] = value
-        self.item.store()
+        with lib.transaction():
+            for field, view in config['import']['set_fields'].items():
+                value = view.get()
+                log.debug(u'Set field {1}={2} for {0}',
+                          displayable_path(self.paths),
+                          field,
+                          value)
+                self.item[field] = value
+            self.item.store()
```

Comment:
The exact same loop as before, wrapped in a transaction.

Reply:
It think there's no need for the transaction here, the only interaction with the database here is in `item.store()` which already uses a transaction internally.

Reply:
thx. fixed.